docs/Learning-Environment-Design-Agents.md

```diff
@@ -47,32 +52,39 @@ method to make decisions which can allow you to control the Agent manually or
 write your own Policy. If the Agent has a `Model` file, its Policy will use the
 neural network `Model` to take decisions.
 
-When you create an Agent, you must extend the base Agent class. This includes
-implementing the following methods:
+When you create an Agent, you should usually extend the base Agent class. This
+includes implementing the following methods:
 
 - `Agent.OnEpisodeBegin()` — Called at the beginning of an Agent's episode,
-  including at the beginning of the simulation. The Ball3DAgent class uses this
-  function to reset the agent cube and ball to their starting positions. The
-  function randomizes the reset values so that the training generalizes to more
-  than a specific starting position and agent cube attitude.
+  including at the beginning of the simulation.
 - `Agent.CollectObservations(VectorSensor sensor)` — Called every simulation
-  step. Responsible for collecting the Agent's observations of the environment.
-  Since the Behavior Parameters of the Agent are set with vector observation
-  space with a state size of 8, the `CollectObservations(VectorSensor sensor)`
```

Comment:
The 3DBall example here sounded really strange when interleaved with the method descriptions. I pulled them out of this part and moved them just below.

Reply:
Wait, I think this line should read "Called every time the Agent requests a decision"

Reply:
Updated.